### PR TITLE
Support `[aria-orientation="vertical"]`

### PR DESCRIPTION
**Pull request by seanpdoyle**
When the `[role="tablist"]` element declares
[`[aria-orientation="vertical"]`][aria-orientation], directional
[keyboard navigation changes][keyboard-interaction] from `ArrowRight`
and `ArrowLeft` to `ArrowDown` and `ArrowUp`:

> When a tab list has its `aria-orientation` set to `vertical`:
> * <kbd>Down Arrow</kbd> performs as <kbd>Right Arrow</kbd> is described above.
> * <kbd>Up Arrow</kbd> performs as <kbd>Left Arrow</kbd> is described above.

[aria-orientation]: https://www.w3.org/TR/wai-aria/#aria-orientation
[keyboard-interaction]: https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/#keyboard-interaction-21

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -15,12 +27,15 @@ export default class TabContainerElement extends HTMLElement {
       if (target.getAttribute('role') !== 'tab' && !target.closest('[role="tablist"]')) return
       const tabs = getTabs(this)
       const currentIndex = tabs.indexOf(tabs.find(tab => tab.matches('[aria-selected="true"]'))!)
+      const [incrementKey, decrementKey] = getNavigationKeys(
+        !!target.closest('[role="tablist"][aria-orientation="vertical"]')
```
Comment (jfuchs) [inline src/index.ts:31]: Would this return `true` if there was a non-vertical tablist nested inside a vertical tablist? And wouldn't you want it to return `false` in that case?

What do you think of something like this?
```suggestion
        !!(target.closest('[role="tablist"]')?.getAttribute('aria-orientation') === 'vertical')
```

If that seems correct, would you mind adding a test for that case? It not, I'm happy to chat about other options (or figure out my misunderstanding)

File: test/test.js [hunk 2]
```diff
@@ -219,4 +219,82 @@ describe('tab-container', function () {
...
+
+      tabs[1].dispatchEvent(new KeyboardEvent('keydown', {code: 'End', bubbles: true}))
+      assert(panels[0].hidden)
+      assert(panels[1].hidden)
+      assert(!panels[2].hidden)
+      assert.equal(document.activeElement, tabs[2])
+      assert.equal(counter, 2)
+    })
+
+    it('does not supports left and right keyboard shortcuts', () => {
```
Comment (jfuchs) [inline test/test.js:278]:
```suggestion
    it('does not support left and right keyboard shortcuts', () => {
```

File: src/index.ts (hunk 1 above)
Comment (seanpdoyle) [reply src/index.ts]: @jfuchs thank you for the review! I understand the implementation change, but I'm unsure about how I'd write a test to nest in the way you're describing. Would you be interested in sharing pseudo code for the example?

**Review by jscholes — CHANGES_REQUESTED** [review]
@seanpdoyle These proposed changes are technically correct according to the ARIA Authoring Practices Guide (APG).  However, that document is not a standard, and unfortunately `aria-orientation` isn't well supported by screen readers.  For instance with NVDA, moving to a `tablist` with  vertical orientation causes exactly the same announcement as a horizontal one.

Add to that, the fact that on Windows, native tab lists are almost always horizontal and/or support horizontal keyboard navigation, and this will cause more confusion than anything else.  User expectation is that Left/Right Arrow will always work, particularly for people unable to see, and I have personally experienced the unnecessary additional cognitive load imposed by strictly following the APG's advice.

My recommendation with a vertical `tablist`, therefore, is to have all four arrows work.  That is, Up and Left Arrow do the same thing as each other, as do Right and Down.  This way, if a sighted keyboard user notices the vertical layout, the keystrokes they expect to press will work.  But for a screen reader user who cannot see it, they can just use it without having to think.  It was also suggested by @jfuchs, via a Slack discussion, that this approach could just be applied to the component as a whole, regardless of orientation.

**Comment by seanpdoyle** [comment]
@jscholes thank you for your review. I've pushed https://github.com/github/tab-container-element/pull/52/commits/e03a83e577d9b0f6e22a5d0792fb1d5f52121b00 to support both sets of keys when `[aria-orientation="vertical"]`.

> It was also suggested by @jfuchs, via a Slack discussion, that this approach could just be applied to the component as a whole, regardless of orientation.

Does that behavior risk further confusion by _always_ intercepting up and down arrow-powered scrolling when a `[role="tablist"]` contains focus?

**Comment by jfuchs** [comment]
> > It was also suggested by @jfuchs, via a Slack discussion, that this approach could just be applied to the component as a whole, regardless of orientation.
> 
> Does that behavior risk further confusion by _always_ intercepting up and down arrow-powered scrolling when a `[role="tablist"]` contains focus?

I'm okay with dropping that plan.

**Comment by jscholes** [comment]
@seanpdoyle Good point.  I'm okay with Up/Down having no function if the orientation is horizontal.

**Review by keithamus — APPROVED** [review]
LGTM :+1:. I think this intersects the concerns nicely. Good implementation, and great work on adding tests!

**Pull request merged**